### PR TITLE
[functorch] fix batching rule for dropout

### LINKED ISSUE #92283: Randomness 'different' results in weird behavior of Dropout
Hi,

I am using functorch for computing Hessian of a network with respect to parameters. In particular, I am doing it for WideResNet and I noticed, that when hessian is computed with 
```
functorch.jacfwd(functorch.jacrev(func), randomness='different')(parameters)
```
 it results in error because somehow, after dropout layer the batch dimension disappears (so all the further layers cannot be applied).
Everything works with randomness='same'

cc @ezyang @gchanan @zou3519 @Chillee @samdow @soumith @kshitij12345 @janeyx99

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes https://github.com/pytorch/pytorch/issues/92283

The repro now works:
```python
import torch
import torch.func
import torch.nn as nn

x = torch.randn(3, device='cuda')
y = torch.randn(1, 3, device='cuda')

def fn(x, y):
    # previously output of dropout used to be incorrect [B, 3] (B=1) and thus `mean(1)` used to fail
    # post the fix output of dropout is [B, 1, 3] and `mean(1)` works.
    return x + nn.functional.dropout(y, 0.3).mean(1)


o = torch.func.vmap(fn, in_dims=(0, None), randomness='different')(x, y)
```

**NOTE**: 
`native_dropout_batching_rule(const Tensor& tensor, double p, c10::optional<bool> train)` was called only for CUDA tensor. Hence this issue only affected CUDA tensors and not CPU tensors

Ref:
https://github.com/pytorch/pytorch/blob/a6ac922eabee8fce7a48dedac81e82ac8cfe9a45/aten/src/ATen/functorch/PyTorchOperatorHacks.cpp#L251-L258

### CODE REVIEW

File: aten/src/ATen/functorch/BatchRulesRandomness.cpp [hunk 1]
```diff
@@ -198,9 +198,16 @@ std::tuple<Tensor,Tensor> native_dropout_batching_rule(const Tensor& tensor, dou
     check_randomness(randomness); // if we are in eval mode, we don't use about randomness
   }
 
-  if ((train.has_value() && !train) || randomness == RandomnessType::Different) {
+  if ((train.has_value() && !train) ||
+      randomness == RandomnessType::Different) {
     auto res = at::native_dropout(tensor_value, p, train);
```
[the remark this answers is not in the document]
Comment (kshitij12345) [reply aten/src/ATen/functorch/BatchRulesRandomness.cpp]: Sure. Thanks!

**Review by Skylion007 — COMMENTED** [review]
Looks good to me now.

**Review by Chillee — APPROVED** [review]

**Review by Skylion007 — APPROVED** [review]
